README.md

```diff
@@ -112,6 +112,43 @@ title,content
 My Title,Hello World!
 ```
 
+### Throttling
+
+Maintenance Tasks often modify a lot of data and can be taxing on your database.
+The gem provides a throttling mechanism that can be used to throttle a Task when
+a given condition is met. Specify the throttle condition as a Proc:
+
+```ruby
+# app/tasks/maintenance/update_posts_throttled_task.rb
+module Maintenance
+  class UpdatePostsThrottledTask < MaintenanceTasks::Task
+    throttle_on -> { DatabaseStatus.unhealthy? }, backoff: 1.minute
+
+    def collection
+      Post.all
+    end
+
+    def count
+      collection.count
+    end
+
+    def process(post)
+      post.update!(content: "New content added on #{Time.now.utc}")
+    end
+  end
+end
+```
+
+Optionally, a `backoff` can be supplied. The default is 30 seconds.
```

Comment:
It's a bit weird that the example uses `backoff` before it's explained. I guess it makes sense to only want one example, but maybe we can leave the details to the API documentation?